Running plan feature

aasssddd opened the pull request.

aasssddd committed:
[WIP]
1. Add running plan fields to LatencyCalibration object

aasssddd committed:
Running plan feature
1. Add running plan test case
2. Add running plan feature to run_load.go

aasssddd committed:
Remove redundant lock variable

tnachen commented on load/run_latency_calibration_test.go:
```diff
@@ -0,0 +1,95 @@
+		err = RunCalibrate(testData)
+		assert.Empty(t, err, "test load data from stdin: %v", err)
+	})
+	// t.Run("Test data from file", func(t *testing.T) {
```
Remove these commented code

tnachen commented on load/run_load.go:
```diff
@@ -160,19 +166,82 @@ func (load *AppLoad) Run() error {
+			Duration:    load.LoadTime,
+		})
+	}
+	// get QPS, Concurrency, LoadTime from Plan
```
Move this before it happens

tnachen commented on load/run_load.go:
```diff
@@ -160,19 +166,82 @@ func (load *AppLoad) Run() error {
+			for {
+				select {
+				case step := <-load.HandlerParams.startStep:
+					fmt.Println("new round")
```
Remove this

tnachen commented on load/run_load.go:
```diff
@@ -160,19 +166,82 @@ func (load *AppLoad) Run() error {
+					// Run Request
+					load.runRequest(&tasks, client)
+				}
+
```
Remove extra line

tnachen commented on load/run_load.go:
```diff
@@ -160,19 +166,82 @@ func (load *AppLoad) Run() error {
+		dstURL, err := url.Parse(tasks[0].UrlTemplate)
+		if err != nil {
+			log.Panicf("Unable to parse url: %v", err.Error())
+		} else {
```
Remove the else, since panic kills the process.

tnachen commented on load/run_load.go:
```diff
@@ -160,19 +166,82 @@ func (load *AppLoad) Run() error {
+	}
+	// get QPS, Concurrency, LoadTime from Plan
+
+	// start runner process
```
Remove this since the method name is pretty obvious

aasssddd committed:
refactoring
1. uncomment test cases
2. remove redundant comments
3. remove redundant lines
4. move code block to proper position
